fault-proof/src/challenger.rs

```diff
@@ -408,13 +497,35 @@ where
             }
         }
 
+        tracing::info!("Challenging job completed");
+
         Ok(())
     }
 
-    pub async fn submit_challenge_transaction(&self, game: &Game) -> Result<()> {
+    pub async fn submit_challenge_transaction(
+        &self,
+        game: &Game,
+        submit_enabled: bool,
+    ) -> Result<()> {
         let contract = OPSuccinctFaultDisputeGame::new(game.address, self.l1_provider.clone());
         let transaction_request =
             contract.challenge().value(self.challenger_bond).into_transaction_request();
+
+        tracing::info!(
+            submit_challenge_tx=submit_enabled,
+            game_index = %game.index,
+            game_address = ?game.address,
+            l2_block = %game.l2_block_number,
+            tx = ?transaction_request,
+            "Challenging game"
+        );
+
+        // If monitor_only mode is enabled, just log and skip challenging
+        if !submit_enabled {
+            tracing::info!("Monitor-only mode enabled, skipping challenge transaction submission. You can submit it manually on-chain if needed.");
+            return Ok(());
+        }
```

Comment:
It'd be more intuitive to swap the logs order.

```suggestion
        // If monitor_only mode is enabled, just log and skip challenging
        if !submit_enabled {
            tracing::info!("Monitor-only mode enabled, skipping challenge transaction submission. You can submit it manually on-chain if needed.");
            return Ok(());
        }
        
        tracing::info!(
            submit_challenge_tx=submit_enabled,
            game_index = %game.index,
            game_address = ?game.address,
            l2_block = %game.l2_block_number,
            tx = ?transaction_request,
            "Challenging game"
        );
```

Reply:
@seolaoh 
Need to log transaction body even in monitor mode so user can sign transaction manually.